pkg/aws/cloud.go

```diff
@@ -47,44 +63,68 @@ func NewCloud(log gwlog.Logger, cfg CloudConfig) (Cloud, error) {
 		}
 	})
 
-	lattice := services.NewDefaultLattice(sess, config.Region)
-	eks := services.NewDefaultEKS(sess, config.Region)
-	cl := NewDefaultCloud(lattice, eks, cfg)
+	lattice := services.NewDefaultLattice(sess, cfg.Region)
+	cl := NewDefaultCloud(lattice, cfg)
 	return cl, nil
 }
 
 // Used in testing and mocks
-func NewDefaultCloud(lattice services.Lattice, eks services.EKS, cfg CloudConfig) Cloud {
-	return &defaultCloud{cfg, lattice, eks}
+func NewDefaultCloud(lattice services.Lattice, cfg CloudConfig) Cloud {
+	return &defaultCloud{
+		cfg:          cfg,
+		lattice:      lattice,
+		managedByTag: getManagedByTag(cfg),
+	}
 }
 
 type defaultCloud struct {
-	cfg     CloudConfig
-	lattice services.Lattice
-	eks     services.EKS
+	cfg          CloudConfig
+	lattice      services.Lattice
+	managedByTag string
 }
 
 func (c *defaultCloud) Lattice() services.Lattice {
 	return c.lattice
 }
 
-func (c *defaultCloud) EKS() services.EKS {
-	return c.eks
-}
-
 func (c *defaultCloud) Config() CloudConfig {
 	return c.cfg
 }
 
-func (d *defaultCloud) GetEKSClusterVPC(name string) string {
-	input := &eks.DescribeClusterInput{
-		Name: aws.String(name),
-	}
+func (c *defaultCloud) DefaultTags() Tags {
+	tags := Tags{}
+	tags[TagManagedBy] = &c.managedByTag
+	return tags
+}
 
-	result, err := d.eks.DescribeCluster(input)
+func (c *defaultCloud) IsManagedByTagSet(tags Tags) bool {
+	tag, ok := tags[TagManagedBy]
+	if !ok || tag == nil {
+		return false
+	}
+	return *tag == c.managedByTag
+}
 
+func (c *defaultCloud) IsArnManaged(arn string) (bool, error) {
+	tagsReq := &vpclattice.ListTagsForResourceInput{ResourceArn: &arn}
+	resp, err := c.lattice.ListTagsForResource(tagsReq)
 	if err != nil {
-		return ""
+		return false, err
+	}
+	isManaged := c.IsManagedByTagSet(resp.Tags)
+	return isManaged, nil
+}
+
+func getManagedByTag(cfg CloudConfig) string {
```

Comment:
if everything is empty the tag value will be empty, do we simply take the value and put in or we check for empty strings? 

seems like its allowed to have tag with value of 0 length (Value Length Constraints: Minimum length of 0. Maximum length of 256.) so maybe fine?